### ISSUE TITLE
Implementation Question

### ISSUE THREAD

**Issue by jrasm91** [ramiroaisen/svelte-material-icons#21]
Is there a specific reason this project decided to pre-generate all the different icons, instead of creating a single icon component, like react did [here](https://github.com/Templarian/MaterialDesign-React/blob/master/src/Icon.tsx)?

**Comment by BogdanCerovac** [comment]
I guess single component would also make fixing #20 much easier...

Waiting for this to be clarified before I make a pull request that fixes #20...

**Comment by BogdanCerovac** [comment]
@jrasm91, I guess the readme has the answer actually;

"This is a simple source to source compiler that transforms the svg files from https://github.com/Templarian/MaterialDesign to svelte components using [cheerio](https://cheerio.js.org/)"

So maybe it could be possible to introduce a simple middleware (or maybe just an addition to mentioned compiler) to add the abstraction similar to your reference...

**Comment by jrasm91** [comment]
Right, but I think I've accomplished the same thing just by using a component, and it seems much simpler.

https://github.com/immich-app/immich/pull/4636/files#diff-602ad35e4bb57980594ed6c80f80c774702c64c984402110a43bd8dff20bfab6

**Comment by BogdanCerovac** [comment]
That looks great as well!

**Comment by jelleromer** [comment]
Sorry for necroposting, but if anyone finds this thread you're probably looking for [Iconify](https://www.npmjs.com/package/@iconify/svelte).
